src/HID-APIs/MultiTouchAPI.h

```diff
@@ -0,0 +1,118 @@
+/*
+Copyright (c) 2021 ilufang
+See the readme for credit to other people.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+// Include guard
+#pragma once
+
+/// Maximum amount of fingers supported
+#define HID_MULTITOUCH_MAXFINGERS 10
+/// Number of fingers in a single report
+#define HID_MULTITOUCH_REPORTFINGERS 2
+
+// A report will always be the same size, even if you report fewer fingers than
+// REPORTFINGERS. The unused finger entries will simply be zero. If more fingers
+// are present than REPORTFINGERS, multiple reports will be sent to report all
+// fingers. This is know as "Hybrid Mode" on MSDN. The number of supported
+// fingers identified by Windows will still be MAXFINGERS. More than MAXFINGERS
+// contacts may be ignored by Windows even with hybrid mode.
+
+#define _MT_STATE_INACTIVE 0
+#define _MT_STATE_CONTACT  1
+#define _MT_STATE_RELEASED 2
+
+typedef struct ATTRIBUTE_PACKED {
+	uint8_t reportID;
+	uint8_t count;
+	struct ATTRIBUTE_PACKED {
+		uint8_t identifier;
+		uint8_t touch;
+		uint8_t pressure;
+		uint8_t x1, x0;
+		uint8_t y1, y0;
+	} contacts[HID_MULTITOUCH_REPORTFINGERS];
+} HID_MultiTouchReport_Data_t;
+
+typedef struct {
+	uint8_t status;
+	int8_t pressure;
+	int16_t x, y;
+} _finger_t;
+
+class MultiTouchAPI
+{
+public:
+	MultiTouchAPI() {
+		_fingers_count = 0;
+		for (int i = 0; i < HID_MULTITOUCH_MAXFINGERS; i++) {
+			_fingers[i].status = _MT_STATE_INACTIVE;
+		}
+	}
```

Comment:
Having the definition separately but in the same unit of translation causes multiple definition problems during linking. But since now all data components are zeros I'll just remove it altogether.

Reply:
Constructor removed because default constructor will correctly initialize all data components.